algorithms/arrays/n_sum.py

```diff
@@ -0,0 +1,97 @@
+"""
+Given an array of n integers, are there elements a, b, .. , n in nums
+such that a + b + .. + n = target?
+
+Find all unique triplets in the array which gives the sum of target.
+
+Example:
+    Given n = 4, nums = [1, 0, -1, 0, -2, 2], target = 0,
+    return [[-2, -1, 1, 2], [-2, 0, 0, 2], [-1, 0, 0, 1]]
+"""
+
+__sum_closure = None
+__compare_closure = None
+
+
+def n_sum(n, nums, target, **kv):
```

Comment:
In this repo, ideally all files have only one function in the global scope (that can be imported). Your implementation is good, but please change it a little bit as follows:
- I think we can drop the optional `keywords arguments`, since almost everytime, the default functions are used. This will reduce some complexity of the code.
- Most of the "hidden" functions `__*` can be just written as is, inside the main function body, specially `__n_sum`.
- If some extra functions are still required, make them as inner functions of `n_sum`.
- Also mention the Time Complexity if the solution. It not easy to follow this code.
Thanks.